src/features/core/deviceModeTransformation/dmtHandler.js

```diff
@@ -0,0 +1,190 @@
+                /**
+                 * Sample Response format:
+                 * {
+                    "transformedBatch" :[
+                      {
+                        "id": "destination-id",
+                        "payload": [
+                          {
+                            "orderNo":1,
+                            "status": "200",
+                            "event": {
+                              "message": { ...}
+                          }]
+                      }]
+                    } 
+                */
+                /**
+                 * Filter the successful transformed event
+                 */
+                const transformationResponse = [];
+                response.transformedBatch.forEach((dest) => {
+                  transformationResponse.push({
+                    id: dest.id,
+                    payload: dest.payload.filter((tEvent) => tEvent.status === '200'),
+                  });
+                });
+
+                resolve({
+                  transformedPayload: transformationResponse,
+                  transformationServerAccess: true,
+                });
+                return;
+              }
+              if (status === 400) {
+                const errorMessage = response
+                  ? `[Transformation]:: ${response}`
+                  : `[Transformation]:: Invalid request payload`;
+                reject(errorMessage);
+                return;
+              }
+              if (status === 404) {
+                resolve({
+                  transformedPayload: payload.batch,
+                  transformationServerAccess: false,
+                });
+                return;
+              }
+
+              // If the request is not successful
+              // one or more transformation is unsuccessfull
+              // retry till the retryAttempt is exhausted
+              if (retryAttempt > 0) {
+                const newretryAttempt = retryAttempt - 1;
+                setTimeout(() => this.sendEventForTransformation(payload, newretryAttempt)
+                    .then(resolve)
+                    .catch(reject), 500);
+              } else {
+                // Even after all the retries event transformation
+                // is not successful, ignore the event
+                reject(`[Transformation]:: Transformation failed with status ${status}`);
+                return;
+              }
+            } catch (err) {
+              reject(err);
+            }
+          }
+        };
+        xhr.send(JSON.stringify(payload, replacer));
+      } catch (error) {
+        reject(error);
+      }
+    });
+  }
+
+  checkQueueLengthAndProcess() {
+    if (this.queue.length > 0) {
+      this.process();
+    }
+  }
+
+  /**
+   * A helper function that will process the transformation
+   * and return the transformed event payload
+   *
+   */
+  process() {
+    this.isTransformationProcessing = true;
+    const firstElement = this.queue.shift();
```

Comment:
i am a bit worried thatw e process the events 1 by 1.

the transformations will probabaly lag a lot compared to the event generation in they take 500ms each

Reply:
discussed offline